build/org.eclipse.cdt.managedbuilder.core/src/org/eclipse/cdt/managedbuilder/internal/core/jsoncdb/generator/CompilationDatabaseGenerator.java

```diff
@@ -229,10 +229,10 @@ private List<CompilationDatabaseInformation> populateObjList(IProject project, I
 						outputLocation + "", inputStrings, sourceLocation, outputLocation); //$NON-NLS-1$
 
 				IBuildMacroProvider provider = ManagedBuildManager.getBuildMacroProvider();
-				String compilerName = CompilationDatabaseGenerator.getCompilerName(tool);
+				String compilerName = CompilationDatabaseGenerator.getCompilerName(tool, config, cmdLInfo);
```

Comment:
Minor change - instead of passing `cmdLInfo` to `getCompilerName`, move this line below `String commandLine = cmdLInfo.getCommandLine();` and pass `commandLine` to `getCompilerName`